awx/main/tasks.py

```diff
@@ -742,10 +739,29 @@ def get_stdout_handle(self, instance):
         '''
         if not os.path.exists(settings.JOBOUTPUT_ROOT):
             os.makedirs(settings.JOBOUTPUT_ROOT)
-        stdout_filename = os.path.join(settings.JOBOUTPUT_ROOT, "%d-%s.out" % (instance.pk, str(uuid.uuid1())))
+        stdout_filename = os.path.join(
+            settings.JOBOUTPUT_ROOT,
+            "%d-%s.out" % (instance.pk, str(uuid.uuid1()))
+        )
         stdout_handle = codecs.open(stdout_filename, 'w', encoding='utf-8')
         assert stdout_handle.name == stdout_filename
-        return stdout_handle
+
+        if getattr(settings, 'USE_CALLBACK_QUEUE', False):
+            dispatcher = CallbackQueueDispatcher()
+
+            def event_callback(event_data):
+                event_data.setdefault(self.event_data_key, instance.id)
+                if 'uuid' in event_data:
+                    cache_event = cache.get('ev-{}'.format(event_data['uuid']), None)
+                    if cache_event is not None:
+                        event_data.update(cache_event)
+                dispatcher.dispatch(event_data)
+        else:
+            def event_callback(event_data):
+                event_data.setdefault(self.event_data_key, instance.id)
+                self.event_model.create_from_data(**event_data)
```

Comment:
I know that you're just moving this from elsewhere, but this doesn't work, does it? The event reconstruction (`event_data.update(cache_event)`) should apply regardless of whether it is saved via the callback receiver or directly. I think the setting `USE_CALLBACK_QUEUE` is not working for this reason, just due to code rot.

Reply:
You mean the `else` case?  It looks like `USE_CALLBACK_QUEUE` defaults to `True` in `settings.defaults`, and isn't set anywhere else (it isn't configurable via the settings API).  I expect this `else` block probably _is_ code rot - maybe we were using this as a mechanism for tests @matburt @cchurch?